lib/auth/tls_test.go

```diff
@@ -1807,6 +1807,107 @@ func TestExtendWebSessionWithReloadUser(t *testing.T) {
 	require.Equal(t, traits[constants.TraitDBUsers], []string{"llama", "alpaca"})
 }
 
+func TestExtendWebSessionWithMaxDuration(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	testSrv := newTestTLSServer(t)
+	clock := testSrv.AuthServer.TestAuthServerConfig.Clock
+
+	clt, err := testSrv.NewClient(TestAdmin())
+	require.NoError(t, err)
+
+	const user = "user2"
+	const testRequestRole = "test-request-role"
+	pass := []byte("abc123")
+
+	newUser, err := CreateUserRoleAndRequestable(clt, user, testRequestRole)
+	require.NoError(t, err)
+	require.Len(t, newUser.GetRoles(), 1)
+
+	require.Len(t, newUser.GetRoles(), 1)
+	require.Empty(t, cmp.Diff(newUser.GetRoles(), []string{"user:user2"}))
+
+	proxy, err := testSrv.NewClient(TestBuiltin(types.RoleProxy))
+	require.NoError(t, err)
+
+	// Create a user to create a web session for.
+	req := AuthenticateUserRequest{
+		Username: user,
+		Pass: &PassCreds{
+			Password: pass,
+		},
+	}
+
+	err = testSrv.Auth().UpsertPassword(user, pass)
+	require.NoError(t, err)
+
+	ws, err := proxy.AuthenticateWebUser(ctx, req)
+	require.NoError(t, err)
+
+	web, err := testSrv.NewClientFromWebSession(ws)
+	require.NoError(t, err)
+
+	testCases := []struct {
+		desc            string
+		maxDurationRole time.Duration
+		expectedExpiry  time.Duration
+	}{
+		{
+			desc:            "default",
+			maxDurationRole: 0,
+			expectedExpiry:  12 * time.Hour, // default certificate TTL
```

Comment:
Is there a constant we could use?

Reply:
Replaced with default cert duration constant. TBH, I'm not sure if using constants in tests is always a good thing. I understand that the readability is better, but if someone decides to change this value (in this case, this is not used only internally), our tests will not fail. I'd expect that changes affecting end users will trigger test failures.

Reply:
I was actually thinking it would make the test more resilient in this case. Readability is arguable - some constants are good at expressing intent, but sometimes I find the actual value clearer.

Reply:
Your call whether you prefer the const or the number.